### PR TITLE
Adding prices and upvoting/downvoting

### PR DESCRIPTION
**Pull request by rsun0**
## Status: 
:rocket: Ready

## Description
Added POST routes for adding prices to existing items and upvoting/downvoting existing prices, and tests for them.

### CODE REVIEW

File: backend/app.py [hunk 1]
```diff
@@ -5,10 +5,26 @@
...
+class Error(Enum):
+    MISSING_FIELDS = 'Must fill all required fields'
+    MISSING_KEYWORD_UPC = 'Request does not contain keyword or upc code'
+    ITEM_DNE = 'Item does not exist in database'
+    STORE_DNE = 'Store does not exist in database'
+    INVALID_DIR = 'Invalid vote direction'
+    ALREADY_UPVOTED = 'User has already upvoted'
+    ALREADY_DOWNVOTED = 'User has already downvoted'
+    NOT_VOTED = 'User has not voted, cannot undo'
+
```
Comment (Pezhin) [inline backend/app.py:20]: Awesome, classical Enum usage 👍

File: backend/app.py [hunk 2]
```diff
@@ -59,6 +75,108 @@ def post_item():
...
+    if item is None:
+        return json.dumps({'success': False, 'error': Error.ITEM_DNE.value})
+
+    loc = {'lat': data['lat'], 'long': data['long']}
+    store = item.stores.filter(name=data['store'], location=loc).first()
+    if store is None:
+        return json.dumps({'success': False, 'error': Error.STORE_DNE.value})
+    else:
+        price = store.prices[-1]
+        if data['dir'] == 1:
```
Comment (Pezhin) [inline backend/app.py]: Consider to abstract 1 and -1 as some Enum values, too? (i.e. VOTE_DIRECTION_UP, you can figure out better name than I do)

After that, this is a perfect case to use `switch-case` to gain better readability and potentially better performance. But python does not support `switch`. :(

File: backend/tests/test_vote.py [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+import copy
+
+
+@pytest.fixture
+def db():
+    connect('mongoenginetest', host='mongomock://localhost')
+    model.Item.objects().delete()
+    yield None
+    model.Item.objects().delete()
+
```
Comment (Pezhin) [inline backend/tests/test_vote.py]: Looks like this kind of helper util functions are copy-pasted in multiple files, maybe move this into a test_util? I could be a little bit bike-shedding here.

File: backend/test_data.py [hunk 4]
```diff
@@ -3,110 +3,110 @@
...
     date=datetime.now().timestamp()
 )
 
 # valid
 price2 = model.Price(
     user='admin',
-    upvote=0,
-    downvote=0,
+    upvotes=users(0),
+    downvotes=users(0, 0),
```
Comment (zclewell) [inline backend/test_data.py]: "users(0)" and "users(0,0)" both return empty array correct? "[]" might be simpler to understand

File: backend/test_data.py [hunk 5]
```diff
@@ -3,110 +3,110 @@
...
+    downvotes=users(19, 20),
     price=12.89,
     date=datetime.now().timestamp()
 )
 
+# valid
+price9 = model.Price(
+    user='testuser',
+    upvotes=users(0),
+    downvotes=users(0, 0),
```
Comment (zclewell) [inline backend/test_data.py]: "users(0)" and "users(0,0)" both return empty array correct? "[]" might be simpler to understand

**Review by zclewell — COMMENTED** [review]
Minor comment

File: backend/tests/test_vote.py (hunk 3 above)
Comment (rsun0) [reply backend/tests/test_vote.py]: I discovered that pytest has a special conftest.py file for sharing fixtures, so thanks for the suggestion!

**Review by zclewell — APPROVED** [review]
Looks good!

**Pull request merged**